### DISCUSSION TITLE
Informations par rapport au sdk

### DISCUSSION THREAD

**Discussion by Nzenou** [capsule-corp-ternoa/ternoa-js#174] [General]
Bonjour,

Je suis actuellement en train de créer une DApp sur Ternoa, mais j'aimerais avoir plus d'informations sur le SDK :

Est-il possible pour un "owner" ou un "creator" d'un NFT de changer les metadata après que ce NFT ait été créé ?
Est-il possible d'assigner un "creator" différent de celui qui exécute/paye la fonction "createNft()" ?
Est-il possible de calculer à l'avance les fees d'un transfert de NFT ?

Merci beaucoup 🙏

Hello,

I am currently developing a DApp on Ternoa, but I would like more information about the SDK:

Is it possible for an owner or a creator of an NFT to change the metadata after the NFT has been created?
Is it possible to assign a different creator from the one who executes/pays for the createNft() function?
Is it possible to calculate in advance the fees for an NFT transfer?
Thank you very much 🙏

**Comment by Victor-Salomon** [comment]
Hi @Nzenou, 
Here are the answers to the three questions (in English as it could benefit other developers having the same questions). 

1. NFT off-chain metadata can't be updated after creation, neither by "owner" nor by "creator". A Basic NFT can be converted to a Capsule NFT or a secret off-chain metadata can be added to the original NFT to become a secret NFT. The main difference between those two is the Secret off-chain metadata **can not** be updated while the Capsule off-chain metadata can be edited. 

2. No, you can't assign transaction fees (+ various other fees) to someone else than "creator" or "owner". You can delegate the NFT to someone else. "Delegatee" could use the NFT without properly owning it and will pay the transaction fee. However, during delegation "owner" can't use the delegated NFT.

3. An NFT transfer does not require any specific fee above the transaction fee (unlike NFT creation which involves some fixed fees on top of the transaction fees). The transaction fee is not a static value it's based on several factors: TRANSACTION WEIGHT + BYTES_LENGTH_FEE of the Signed transaction hash. You should be able to test the following method from Polkadot on your unsigned transaction directly : 
  - const txInfo = await tx.paymentInfo(// the singer polkadot address)
  - const estimatedFee = txInfo.partialFee.toString()
  
Enjoy building with Ternoa-js :sparkles:
